mode_dark_light.py

```diff
@@ -2430,12 +2463,29 @@ def create_kernel_pannel(master):
 LINK_LENGTH = 40
 GRID_WIDTH = (TILE_WIDTH + LINK_LENGTH) * COLS - LINK_LENGTH
 GRID_HEIGHT = (TILE_HEIGHT + LINK_LENGTH) * ROWS - LINK_LENGTH
-create_kernel_pannel(master)
-create_mapping_pannel(master)
-create_cgra_pannel(master, ROWS, COLS)
-create_param_pannel(master)
-create_test_pannel(master)
-create_layout_pannel(master)
+
+#set size first to avoid window keep resizing during loading
+w, h = master.winfo_screenwidth(), master.winfo_screenheight()
+master.geometry("%dx%d" % (w-10, h-70))
+master.geometry("+%d+%d" % (0, 0))
+
+main_frame = customtkinter.CTkFrame(master)
+
+overlay = customtkinter.CTkToplevel(master)  
+overlay.geometry("%dx%d" % (w-10, h-70))
+overlay.transient(master)  
+overlay.grab_set()  
+
+loading_label = customtkinter.CTkLabel(overlay, text="Loading...", font=("Arial", 24, "bold"))
+loading_label.place(relx=0.5, rely=0.4, anchor="center")
+
+progress = customtkinter.CTkProgressBar(overlay)
+progress.place(relx=0.5, rely=0.5, anchor="center")
+progress.start()
+
+#adding other UI components in a separate thread

```

Comment:
`# Adds other UI components in a separate thread.`

Reply:
done